cloudify_cli/commands/agents.py

```diff
@@ -120,185 +103,146 @@ def install(deployment,
-                    .format(workflow_id, deployment_id))
-
-    # install agents for all deployments
-    else:
-        dep_list = [dep.id for dep in
-                    client.deployments.list()
-                    if _is_deployment_installed(client, dep.id)]
-        if not dep_list:
-            error_msg = 'There are no deployments installed'
-            return dep_list, error_msg
-
-        logger.info("Running workflow '{0}' for all installed deployments".
-                    format(workflow_id))
-
-    return dep_list, error_msg
-
-
-def run_worker(
-        deps, client, logger, include_logs, workflow_id, parameters=None):
-
-    error_summary = []
-    error_summary_lock = threading.Lock()
-    event_lock = threading.Lock()
-
-    def log_to_summary(message):
-        with error_summary_lock:
-            error_summary.append(message)
-
-    def threadsafe_log(message):
-        with event_lock:
-            logger.info(message)
-
-    def threadsafe_events_logger(events):
-        with event_lock:
-            for event in events:
-                output = logs.create_event_message_prefix(event)
-                if output:
-                    logger.info(output)
-
-    def worker(dep_id):
-        timeout = 900
-        try:
-            execution = client.executions.start(
-                dep_id, workflow_id, parameters,
+    tenants_to_ni_cache = get_node_instances_map(
+        client, agent_filters, all_tenants)
+
+    if not tenants_to_ni_cache:
+        raise CloudifyCliError("No eligible deployments found")
+
+    for tenant_name, node_instances in tenants_to_ni_cache.items():
+        tenant_client = env.get_rest_client(tenant_name=tenant_name)
+        # Group node instances by deployment ID's.
+        deployments_map = dict()
+        for node_instance in node_instances:
+            dep_instances = deployments_map.setdefault(
+                node_instance.deployment_id, list())
+            dep_instances.append(node_instance)
+
+        for deployment_id, dep_node_instances in deployments_map.items():
+            execution_params = {
+                'node_instance_ids': [ni.id for ni in dep_node_instances],
+            }
+            if agent_filters[cfy.AGENT_FILTER_INSTALL_METHODS]:
+                execution_params['install_methods'] = agent_filters[
+                    cfy.AGENT_FILTER_INSTALL_METHODS]
+            if parameters:
+                execution_params.update(parameters)
+            execution = tenant_client.executions.start(
+                deployment_id, workflow_id, execution_params,
                 allow_custom_parameters=True)
-            execution = wait_for_execution(
-                client,
-                execution,
-                events_handler=threadsafe_events_logger,
-                include_logs=include_logs,
-                timeout=timeout
+            logger.info(
+                "Scheduled execution for deployment '%s': %s",
```

Comment:
why "scheduled" and not "started"?

Reply:
@idob-cloudify committed a change for that.